pkg/internal/apis/certmanager/validation/certificate.go

```diff
@@ -79,13 +79,19 @@ func ValidateCertificateSpec(crt *internalcmapi.CertificateSpec, fldPath *field.
 	if crt.Duration != nil || crt.RenewBefore != nil {
 		el = append(el, ValidateDuration(crt, fldPath)...)
 	}
+
 	if len(crt.Usages) > 0 {
 		el = append(el, validateUsages(crt, fldPath)...)
 	}
+
 	if crt.RevisionHistoryLimit != nil && *crt.RevisionHistoryLimit < 1 {
 		el = append(el, field.Invalid(fldPath.Child("revisionHistoryLimit"), *crt.RevisionHistoryLimit, "must not be less than 1"))
 	}
 
+	if crt.MaxPathLen != nil && !crt.IsCA {
+		el = append(el, field.Required(fldPath.Child("isCA"), "must set isCA to true when maxPathLen is set"))
+	}
+
```

Comment:
Please add validation that pathlen may only be a value of nil or 0 and higher.

Reply:
good spot, totally missed that. thanks :+1: